src/pages/patch/BuildVariants.tsx

```diff
@@ -35,27 +38,61 @@ export const BuildVariants: React.FC = () => {
         <Skeleton active title={false} paragraph={{ rows: 4 }} />
       )}
       {data &&
-        data.patchBuildVariants.map(({ displayName, tasks }) => (
+        data.patchBuildVariants.map(({ displayName, tasks, variant }) => (
           <BuildVariant
             key={`buildVariant_${displayName}`}
-            className="patch-build-variant"
+            data-cy="patch-build-variant"
           >
-            <P1>{displayName}</P1>
-            <VariantTasks>
-              {tasks.map((task) => (
-                <TaskSquare key={task.id} {...task} />
-              ))}
-            </VariantTasks>
+            <P1>
+              <Link to={`${getVersionRoute(id)}?page=0&variant=${variant}`}>
+                {displayName}
+              </Link>
+            </P1>
+            <VariantTaskGroup variant={variant} tasks={tasks} />
           </BuildVariant>
         ))}
     </SiderCard>
   );
 };
 
+interface VariantTaskGroupProps {
+  tasks: PatchBuildVariantTask[];
+  variant: string;
+}
+const VariantTaskGroup: React.FC<VariantTaskGroupProps> = ({
+  tasks,
+  variant,
+}) => {
+  const groupedTasks = groupTasksByStatus(tasks);
+  return (
+    <VariantTasks>
+      {Object.keys(groupedTasks).map((status) => (
+        <GroupedTaskSquare
+          key={`${variant}_${status}`}
+          status={status}
+          count={groupedTasks[status]}
+        />
+      ))}
+    </VariantTasks>
+  );
+};
+
 const BuildVariant = styled.div`
   margin-bottom: 8px;
 `;
 const VariantTasks = styled.div`
   display: flex;
   flex-wrap: wrap;
 `;
+
+const groupTasksByStatus = (tasks: PatchBuildVariantTask[]) => {
```

Comment:
I added tooltips that describe the status of each square. You are also able to click on a square to filter just by the status. It might be confusing if 1 square now relates to 3 different statuses